services/src/main/java/org/keycloak/broker/saml/SAMLEndpoint.java

```diff
@@ -404,6 +412,17 @@ private String getEntityId(UriInfo uriInfo, RealmModel realm) {
                 return configEntityId;
         }
 
+        private List<KeyWrapper> getEncryptionKeys() {
+            if (Boolean.getBoolean("keycloak.saml.backward.compatible.encryption")) {
+                KeyManager.ActiveRsaKey keys = session.keys().getActiveRsaKey(realm);
+            }
+            return session.keys()
+                    .getKeysStream(realm, KeyUse.ENC, JWEConstants.RSA_OAEP)
+                    // Sort keys so the active key is the first one
+                    .sorted(SamlService::compareKeys)
```

Comment:
If for some reason multiple keys are marked as active we use priority to pick the best one. Is this considered here?

Is this method only used to decrypt assertions? If this is also used to encrypt assertions this would result in passive keys potentially being used to encrypt assertions.

I would suggest renaming this to getDecryptionKeys perhaps?

Reply:
For SAML we never encrypt using realm keys. We encrypt by a key provided by the client/SP.